### ISSUE TITLE
Atlas Migrate Diff - In Golang with Gorm

### ISSUE THREAD

**Issue by painh** [ariga/atlas#2728]
Hello.

Please help from the community on the hcl configuration for multiple schemes.

I'm using machine translation, so please understand if the content is weird.

```
data "external_schema" "gorm" {
program = [
"go",
"run",
"-mod=mod",
"./loader",
]
}

env "gorm" {
src = data.external_schema.gorm.url
// // dev = "mysql://root:rootpassword@localhost:3306/"
dev = "docker://mysql/8/"
migration {
dir = "file://migrations"
}
format {
migrate {
diff = "{{ sql . \" \" }}"
}
}
}
```
Excluding /dev from the dev= entry causes problems.

```
Error: read state from "gorm/schema.sql": executing statement: "CREATE TABLE `user_management`.`users` (`user_id` bigint unsigned AUTO_INCREMENT,`username` varchar(255) NOT NULL,`email` varchar(255) NOT NULL,`created_at` datetime(3) NULL,`updated_at` datetime(3) NULL,PRIMARY KEY (`user_id`),CONSTRAINT `uni_user_management_users_username` UNIQUE (`username`),CONSTRAINT `uni_user_management_users_email` UNIQUE (`email`));": Error 1049 (42000): Unknown database 'user_management'
```
Error that the database could not be found.

```
type User struct {
UserID uint `gorm:"primaryKey"`
Username string `gorm:"size:255;not null;unique"`
Email string `gorm:"size:255;not null;unique"`
CreatedAt time.Time `gorm:"autoCreateTime"`
UpdatedAt time.Time `gorm:"autoUpdateTime"`
}

func (User) TableName() string {
return "user_management.users"
}
```
I am implementing TableName to return even the scheme name.

dev = "docker://mysql/8/"

I wonder exactly what account is logged in to.
The user_management naturally exists on db, and if you remove the scheme name and work as a single,

```
atlas migrate diff --env gorm
```

The command works well.

I wonder if it is possible to work on multischeme(database).

To sum up,

1. Account to use when using docker option
2. Settings for Multischeme
3. Whether the scheme.table format is available

Thank you.

**Comment by masseelch** [comment]
Your SQL is bound to a database. The dev-database you are starting with the dev-url is not spinning up the database. If there is only one schema: 

```hcl
env "gorm" {
  ....
  dev = "docker://mysql/8/user_management"
  ...
}
```

If there is more then one, use the baseline for the dev database: https://atlasgo.io/concepts/dev-database#baseline-schema

**Comment by painh** [comment]
@masseelch Thank you, it's been resolved, have a nice day.
